### PR TITLE
feat(wow): align snapshot aggregation query API

### PR DESCRIPTION
**Pull request by Ahoo-Wang**
## Description

Align `@ahoo-wang/fetcher-wow` with the snapshot elements aggregation API introduced by upstream Wow [#3017](https://github.com/Ahoo-Wang/Wow/pull/3017).

- add the public TypeScript `AggregationQuery` wire contract, discriminated group/metric unions, and related enums
- add typed JSON `aggregate<Row>()` and SSE `aggregateStream<Row>()` methods to the existing `SnapshotQueryClient`
- reuse the existing `snapshot/aggregation` route, Fetcher decorators, abort handling, and SSE extractor
- add runtime wire-contract tests and compile-time boundary/return-type tests

`SnapshotQueryApi` remains the single public query interface. Its two new aggregation declarations are optional to preserve source compatibility for existing structural mocks/adapters, while the concrete `SnapshotQueryClient` always implements both methods.

The generator is intentionally unchanged; it already consumes the current Wow query-field metadata contract. No dependencies were added.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?

- [x] `pnpm --filter @ahoo-wang/fetcher-wow test` — 23 files, 301 tests, plus TypeScript type checks
- [x] `pnpm --filter @ahoo-wang/fetcher-wow build`
- [x] `pnpm test:unit` — all 12 package suites passed
- [x] `git diff --check`

**Test Configuration**:

- Node.js: repository-supported Node.js >= 18.20.8
- pnpm: 10.34.5
- Platform: macOS

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

Lint completes with zero errors and 72 warnings: 68 baseline warnings plus 4 new `no-explicit-any` warnings from the approved, existing-style `Record<string, any>` attributes on the two new client methods and their interface declarations.

### CODE REVIEW

**Comment by Ahoo-Wang** [comment]
@codex review

File: packages/wow/src/query/index.ts [hunk 1]
```diff
@@ -12,6 +12,7 @@
  */
 
 export * from './condition';
+export * from './aggregation';
```
[the remark this answers is not in the document]
Comment (Ahoo-Wang) [reply packages/wow/src/query/index.ts:15]: Keeping this PR scoped to `packages/wow` per the explicit approved requirement. Syncing `skills/fetcher-wow-cqrs/references/api.md` is intentionally deferred to a separately authorized change, so this thread is resolved as out of scope.

File: packages/wow/src/query/snapshot/snapshotQueryApi.ts [hunk 2]
```diff
@@ -31,6 +33,20 @@ export interface SnapshotQueryApi<
   S,
   FIELDS extends string = string,
 > extends QueryApi<MaterializedSnapshot<S>, FIELDS> {
+  /** Runs a snapshot aggregation and returns all result rows. */
+  aggregate<Row extends DynamicDocument = DynamicDocument>(
+    query: AggregationQuery<FIELDS>,
+    attributes?: Record<string, any>,
+    abortController?: AbortController,
+  ): Promise<Row[]>;
```
[the remark this answers is not in the document]
Comment (Ahoo-Wang) [reply packages/wow/src/query/snapshot/snapshotQueryApi.ts]: Fixed in `9afac3c8`. Per the author decision, the API remains a single `SnapshotQueryApi`; `aggregate` and `aggregateStream` are optional interface members so existing structural mocks/adapters remain source-compatible, while `SnapshotQueryClient` continues to provide both concrete methods.

**Comment by Ahoo-Wang** [comment]
@codex review
